clients/command-kafka/src/main/java/org/eclipse/hono/client/command/kafka/KafkaBasedInternalCommandSender.java

```diff
@@ -69,17 +70,28 @@ public KafkaBasedInternalCommandSender(final KafkaProducerFactory<String, Buffer
             throw new IllegalArgumentException("command is not an instance of KafkaBasedCommand");
         }
 
+        final String topicName = getInternalCommandTopic(adapterInstanceId);
+        final Span currentSpan = startChildSpan(
+                "delegate Command request",
+                topicName,
+                command.getTenant(),
+                command.getDeviceId(),
+                commandContext.getTracingContext());
+
         return sendAndWaitForOutcome(
-                getInternalCommandTopic(adapterInstanceId),
+                topicName,
                 command.getTenant(),
                 command.getDeviceId(),
                 command.getPayload(),
                 getHeaders((KafkaBasedCommand) command),
```

Comment:
In this PR, the `KafkaBasedCommandResponseSender.getHeaders(..)` now return `Map<String, Object>` instead of `List<KafkaHeader>`. The same we can also apply in this class too to. This is just a minor side note.

Reply:
We could, but in this case we are only forwarding an existing command message to the internal topic and we can therefore take advantage of simply copying the existing headers from the incoming message to the outgoing message, i.e. no need to create a Map and re-encode the outgoing Kafka headers again.